### PR TITLE
FSET-963 Reschedule a new video interview when adjustments change

### PR DESCRIPTION
**Pull request by mgilgar**
Bug fix

### CODE REVIEW

File: app/services/onlinetesting/Phase3TestService.scala [hunk 1]
```diff
@@ -185,70 +186,80 @@ trait Phase3TestService extends OnlineTestService with Phase3TestConcern {
...
-        gatewayConfig.phase3Tests.invigilatedTimeToExpireInDays - daysUntilExpiry,
-        "InvigilatedInviteSystem"
-      )
-    } else {
-      Future.successful(())
+    def extendIfInvigilatedOrIfAdjustmentsWereChanged(application: OnlineTestApplication): Future[Unit] = {
+      def couldHaveBeenInvigilatedBefore(phase3TestGroup: Option[Phase3TestGroup]): Boolean =
+        phase3TestGroup.map(_.expirationDate.isAfter(dateTimeFactory.nowLocalTimeZone.plusDays(daysUntilExpiry*2))).getOrElse(false)
+
+      if (couldHaveBeenInvigilatedBefore(phase3TestGroup)) {
```
Comment (henricook) [inline app/services/onlinetesting/Phase3TestService.scala]: I think this can probably be condensed into one if {} elseif {} else {}

Comment (mgilgar) [inline app/services/onlinetesting/Phase3TestService.scala]: It is a bit verbose but I think if you want to understand the code, it is clearer to separate the four cases. I thought about a match but I think it reads better like this.
If I do not convince you, I will change it.

Comment (henricook) [inline app/services/onlinetesting/Phase3TestService.scala]: I think something like this ends up just as readable, and saves the repetition of future.success

      if (couldHaveBeenInvigilatedBefore && !application.isInvigilatedVideo) {
          extendTestGroupExpiryTime(
            application.applicationId,
            - gatewayConfig.phase3Tests.invigilatedTimeToExpireInDays + daysUntilExpiry,
            "NonInvigilatedInviteSystem"
          )
      } else if (!couldHaveBeenInvigilatedBefore && application.isInvigilatedVideo) {
          extendTestGroupExpiryTime(
            application.applicationId,
            gatewayConfig.phase3Tests.invigilatedTimeToExpireInDays - daysUntilExpiry,
            "InvigilatedInviteSystem"
          )
      } else {
        Future.successful(())
      }

Comment (mgilgar) [inline app/services/onlinetesting/Phase3TestService.scala]: I have made the change. Let me know if there is anything else.

**Pull request merged**